systemtest/src/test/java/io/strimzi/systemtest/upgrade/AbstractUpgradeST.java

```diff
@@ -91,49 +92,50 @@ public class AbstractUpgradeST extends AbstractST {
     // ExpectedTopicCount contains additionally consumer-offset topic, my-topic and continuous-topic
     protected final int expectedTopicCount = upgradeTopicCount + 3;
 
-    public static final String UPGRADE_JSON_FILE = "StrimziUpgradeST.json";
-    public static final String DOWNGRADE_JSON_FILE = "StrimziDowngradeST.json";
+    public static final String UPGRADE_YAML_FILE = "StrimziUpgradeST.yaml";
+    public static final String DOWNGRADE_YAML_FILE = "StrimziDowngradeST.yaml";
 
     protected File kafkaYaml;
 
-    protected static JsonArray readUpgradeJson(String fileName) {
+    public static List<VersionModificationData> getVersionModificationData(String filename) {
         try {
-            String jsonStr = IOUtils.toString(new FileReader(TestUtils.USER_PATH + "/src/test/resources/upgrade/" + fileName));
+            ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+            CollectionType modificationDataListType = mapper.getTypeFactory().constructCollectionType(List.class, VersionModificationData.class);
+            List<VersionModificationData> versionModificationDataList = mapper.readValue(new File(TestUtils.USER_PATH + "/src/test/resources/upgrade/" + filename), modificationDataListType);
 
-            return new JsonArray(jsonStr);
-        } catch (IOException e) {
-            e.printStackTrace();
-            throw new RuntimeException(TestUtils.USER_PATH + "/src/test/resources/upgrade/" + fileName + " file was not found.");
+            return versionModificationDataList;
+        } catch (Exception ex) {
+            LOGGER.error("Error while parsing ST data from YAML ", ex);
         }
+        return null;
```

Comment:
You should throw exception instead of returning `null` -> the test will continue even if it will not find the yaml and it will fail further for NPE